hyper/http11/connection.py

```diff
@@ -137,6 +144,33 @@ def connect(self):
 
         return
 
+    @classmethod
+    def _create_tunnel(cls, proxy_host, proxy_port, target_host, target_port):
+        """
+        Sends CONNECT method to a proxy and returns a socket with established
+        connection to the target.
+
+        :returns: socket
+        """
+        conn = cls(proxy_host, proxy_port)
+        conn.connect()
+        conn._send_headers(to_bytestring('CONNECT'),
+                           to_bytestring('%s:%d' % (target_host, target_port)),
+                           # TODO proxy-authorization headers
+                           headers=HTTPHeaderMap())
+
+        response = None
+        while response is None:
+            # 'encourage' the socket to receive data.
+            conn._sock.fill()
+            response = conn.parser.parse_response(conn._sock.buffer)
```

Comment:
Is there any reason we can't just actually use the regular request/response logic, and just not preload any data?